homeassistant/components/switchbot_cloud/light.py

```diff
@@ -0,0 +1,183 @@
+            f"self._attr_brightness = {self._attr_brightness},"
+            f"self._attr_rgb_color = {self._attr_rgb_color},"
+            f"self._attr_color_temp_kelvin = {self._attr_color_temp_kelvin}"
+        )
+        brightness: int | None = kwargs.get("brightness")
+        rgb_color: tuple[int, int, int] | None = kwargs.get("rgb_color")
+        color_temp_kelvin: int | None = kwargs.get("color_temp_kelvin")
+        _LOGGER.info(
+            f"before async_turn_on "
+            f"brightness = {brightness},"
+            f"rgb_color = {rgb_color},"
+            f"color_temp_kelvin = {color_temp_kelvin}"
+        )
+        if brightness is not None:
+            self._attr_color_mode = ColorMode.RGB
+            await self.send_api_command(
+                RGBWWLightCommands.SET_BRIGHTNESS,
+                parameters=str(value_map_brightness(brightness)),
+            )
+            self._attr_brightness = brightness
+        if rgb_color is not None:
+            self._attr_color_mode = ColorMode.RGB
+            await self.send_api_command(
+                RGBWWLightCommands.SET_COLOR,
+                parameters=":".join([str(i) for i in rgb_color]),
+            )
+            self._attr_rgb_color = rgb_color
+        if color_temp_kelvin is not None:
+            self._attr_color_mode = ColorMode.COLOR_TEMP
+            await self.send_api_command(
+                RGBWWLightCommands.SET_COLOR_TEMPERATURE,
+                parameters=str(color_temp_kelvin),
+            )
+            self._attr_color_temp_kelvin = color_temp_kelvin
+            self._attr_rgb_color = (0, 0, 0)
+        else:
+            self._attr_color_mode = ColorMode.RGB
+            await self.send_api_command(CommonCommands.ON)
+        self._attr_is_on = True
+        _LOGGER.info(
+            f"after async_turn_on self._attr_is_on = {self._attr_is_on},"
+            f"self._attr_brightness = {self._attr_brightness},"
+            f"self._attr_rgb_color = {self._attr_rgb_color},"
+            f"self._attr_color_temp_kelvin = {self._attr_color_temp_kelvin}"
+        )
+        self.async_write_ha_state()
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of this light between 0..255."""
+        return self._attr_brightness
+
+    @property
+    def color_mode(self) -> ColorMode | str | None:
+        """Return the color mode of the light."""
+        return self._attr_color_mode
+
+    @property
+    def color_temp_kelvin(self) -> int | None:
+        """Return the CT color value in Kelvin."""
+        return self._attr_color_temp_kelvin
+
+    @property
+    def rgb_color(self) -> tuple[int, int, int] | None:
+        """Return the rgb color value [int, int, int]."""
+        return self._attr_rgb_color
+
+    @property
+    def supported_color_modes(self) -> set[ColorMode] | set[str] | None:
+        """Flag supported color modes."""
+        return self._attr_supported_color_modes
```

Comment:
The default behavior for all these properties is to return their shorthand counterpart, so this is not needed